# The below script is for the following case study - 

There is 1 dataset(csv) with 3 years worth of customer orders. There are 4 columns in the csv dataset: index, CUSTOMER_EMAIL(unique identifier as hash), Net_Revenue, and Year.
For each year we need the following information:
* Total revenue for the current year
* New Customer Revenue e.g. new customers not present in previous year only
* Existing Customer Growth. To calculate this, use the Revenue of existing customers for current year –(minus) Revenue of existing customers from the previous year
* Revenue lost from attrition
* Existing Customer Revenue Current Year
* Existing Customer Revenue Prior Year
* Total Customers Current Year
* Total Customers Previous Year
* New Customers
* Lost Customers
Additionally, generate a few unique plots highlighting some information from the dataset. Are there any interesting observations?

Dataset
https://www.dropbox.com/sh/xhy2fzjdvg3ykhy/AADAVKH9tgD_dWh6TZtOd34ia?dl=0
customer_orders.csv

Output
An HTML website with the results of the data. Please highlight which year the calculations are for. All code should be hosted on GitHub for viewing. Please provide URL’s to both the output and the GitHub repo.

* If you submit a jupyter notebook, also submit the accompanying python file. You may use python(.py), R, and RMD(knit to HTML) files. Other languages are acceptable as well.


## Importing necessary packages

In [ ]:
import pandas as pd

# Uploading the data file using pandas. The data is in the "data" variable

In [ ]:
from google.colab import files
uploaded = files.upload()
import io
data = pd.read_csv(io.BytesIO(uploaded['casestudy.csv']))

Saving casestudy.csv to casestudy.csv


In [ ]:
data.head()

Unnamed: 0         customer_email  net_revenue  year
0           0   nhknapwsbx@gmail.com       249.92  2015
1           1   joiuzbvcpn@gmail.com        87.61  2015
2           2   ukkjctepxt@gmail.com       168.38  2015
3           3   gykatilzrt@gmail.com        62.40  2015
4           4   mmsgsrtxah@gmail.com        43.08  2015

## Total revenue for the current year

The below code displays the total revenue for each year

Answer: the revenue for 2017(the last year in the dataset) is: 31417495.03



In [ ]:
revenue_per_year=data.groupby('year')['net_revenue'].sum()
print(revenue_per_year)

year
2015    29036749.19
2016    25730943.59
2017    31417495.03
Name: net_revenue, dtype: float64


## New Customer Revenue e.g. new customers not present in previous year only

Splitting the dataframe based on year

In [ ]:
data_in_2015=data[data['year']==2015]
data_in_2016=data[data['year']==2016]
data_in_2017=data[data['year']==2017]

<class 'pandas.core.frame.DataFrame'>
Int64Index: 231294 entries, 0 to 231293
Data columns (total 4 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Unnamed: 0      231294 non-null  int64  
 1   customer_email  231294 non-null  object 
 2   net_revenue     231294 non-null  float64
 3   year            231294 non-null  int64  
dtypes: float64(1), int64(2), object(1)
memory usage: 8.8+ MB
None
<class 'pandas.core.frame.DataFrame'>
Int64Index: 204646 entries, 231294 to 435939
Data columns (total 4 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Unnamed: 0      204646 non-null  int64  
 1   customer_email  204646 non-null  object 
 2   net_revenue     204646 non-null  float64
 3   year            204646 non-null  int64  
dtypes: float64(1), int64(2), object(1)
memory usage: 7.8+ MB
None
<class 'pandas.core.frame.DataFrame'>
Int64Index: 249987 entries, 435940 to 685926
Data c

### Extracting the email id of each customer in each year

In [ ]:
cust_in_2015=set(data_in_2015['customer_email'])
cust_in_2016=set(data_in_2016['customer_email'])
cust_in_2017=set(data_in_2017['customer_email'])


### Extracting list of new customer in 2016

In [ ]:
temp_2016=list(set(cust_in_2016) - set(cust_in_2015))

145062


### Obtaining revenue of new customers in 2016

In [ ]:
temp_customer_2016=0
for i in range(0,len(temp_2016)):
  var=temp_2016[i]
  temp_customer_2016=temp_customer_2016+data_in_2016[data_in_2016['customer_email']==var]['net_revenue'].sum()
print(temp_customer_2016)


18245491.010000154


### Extracting list of new customer in 2017

In [ ]:
temp_2017=list(set(cust_in_2017) - set(cust_in_2016))
print(len(temp_2017))

229028


### Obtaining revenue of new customers in 2017

In [ ]:
temp_customer_2017=0
for i in range(0,len(temp_2017)):
  var=temp_2017[i]
  temp_customer_2017=temp_customer_2017+data_in_2017[data_in_2017['customer_email']==var]['net_revenue'].sum()
print(temp_customer_2017)

28776235.039999083


### Calculating total revenue per year

In [ ]:
total_revenue_data_in_2015=data_in_2015['net_revenue'].sum()
total_revenue_data_in_2016=data_in_2016['net_revenue'].sum()
total_revenue_data_in_2017=data_in_2017['net_revenue'].sum()

### Calculating existing revenue per year(total_revenue for that year - revenue by new customers that year)

In [ ]:
existing_revenue_data_in_2015=total_revenue_data_in_2015
existing_revenue_data_in_2016=total_revenue_data_in_2016-temp_customer_2016
existing_revenue_data_in_2017=total_revenue_data_in_2017-temp_customer_2017

### Calculating growth per year(existing revenue of that year - existing revenue of previous year)

In [ ]:
existing_growth_2016=existing_revenue_data_in_2016-existing_revenue_data_in_2015
existing_growth_2017=existing_revenue_data_in_2017-existing_revenue_data_in_2016

### Calculating total customers per year
Total Customers Current Year(2017): 249987

Total Customers Previous Year(2016): 204646

In [ ]:
# Total Customers 2015
print(len(cust_in_2015))

# Total Customers 2016
print(len(cust_in_2016))

# Total Customers 2017
print(len(cust_in_2017))


231294
204646
249987


### List of new customers in 2016

In [ ]:
# New customers in 2016
for i in cust_in_2016:
  if i not in cust_in_2015:
    print(i)


### List of new customers in 2017

In [ ]:
# New customers in 2017
for i in cust_in_2017:
  if i not in cust_in_2016:
    print(i)

### List of Customers lost in 2016

In [ ]:
# LOST In 2016
for i in cust_in_2015:
  if i not in cust_in_2016:
    print(i)


### List of Customers lost in 2017

In [ ]:
# LOST In 2017
for i in cust_in_2016:
  if i not in cust_in_2017:
    print(i)